skimage/transform/radon_transform.py

```diff
@@ -18,23 +19,27 @@
 __all__ = ['radon', 'order_angles_golden_ratio', 'iradon', 'iradon_sart']
 
 
-def radon(image, theta=None, circle=True):
+def radon(image, theta=None, circle=True, preserve_range=False):
```

Comment:
To keep the old behavior going, this should be set to `None` with a warning for users that it will change in a future version.

@jni, this function's documentation only included support for floating point types, but actually accepted integer types. Should we follow the general deprecation procedure?

Reply:
Actually, setting `preserve_range` to `True` keeps the actual behavior unchanged since with the actual version of `convert_to_float` directly convert any array to `double` when `preserve_range` is `True`.
When `preserve_range` is False, `img_as_float` is called, and in this case, non float arrays are rescaled betweeen 0 and 1. The backward compatibility may be broken here.
Should we raise a deprecation warning when `preserve_range` is `False` and `dtype` is not floating?

Reply:
Right, so I was suggesting we go through a deprecation cycle.

1. Set `preserve_range=None` by default.
2. If not set, and we have integer types, raise a warning, and set it to True
3. In 0.18, make it `False` by default.